Common/cpp/SharedItems/Shareables.cpp

```diff
@@ -245,13 +246,16 @@ jsi::Value ShareableRemoteFunction::toJSValue(jsi::Runtime &rt) {
 }
 
 jsi::Value ShareableHandle::toJSValue(jsi::Runtime &rt) {
-  if (initializer_ != nullptr) {
+  if (remoteValue_ == nullptr) {
     auto initObj = initializer_->getJSValue(rt);
-    remoteValue_ =
-        std::make_unique<jsi::Value>(getValueUnpacker(rt).call(rt, initObj));
-    remoteRuntime_ = &rt;
-    initializer_ = nullptr; // we can release ref to initializer as this
-    // method should be called at most once
+    auto value = std::make_unique<jsi::Value>(getValueUnpacker(rt).call(
+        rt, initObj, jsi::String::createFromAscii(rt, "Handle")));
+
+    std::unique_lock<std::mutex> lock(initializationMutex_, std::try_to_lock);
```

Comment:
Wy need to remove `std::try_to_lock` because this will lead to calling `*remoteValue_` on `null` :/

Reply:
smart